Sources/AppKit/NSTableView.swift

```diff
@@ -36,20 +37,164 @@ public extension ReactiveExtensions where Base: NSTableView {
   }
 }
 
-public extension SignalProtocol where Element: DataSourceEventProtocol, Element.DataSource: QueryableDataSourceProtocol, Element.DataSource.Item: Any, Element.DataSource.Index == Int, Error == NoError {
+// MARK: - Table view integration
+
+extension NSTableView {
+	
+	fileprivate var updating: Bool {
+		get {
+			return objc_getAssociatedObject(self, &NSTableViewUpdatingKey) as? Bool ?? false
+		}
+		set {
+			objc_setAssociatedObject(self, &NSTableViewUpdatingKey, newValue, .OBJC_ASSOCIATION_RETAIN)
+		}
+	}
+}
 
-  public typealias DataSource = Element.DataSource
+private var NSTableViewUpdatingKey: UInt8 = 0
+
+// MARK: - Bond declaration
+
+public protocol TableViewBondOptionable {
+	var insertAnimation: NSTableViewAnimationOptions? { get }
+	var deleteAnimation: NSTableViewAnimationOptions? { get }
+}
 
-  @discardableResult
-  public func bind(to tableView: NSTableView, animated: Bool = true, createCell: @escaping (DataSource, Int, NSTableView) -> NSView?) -> Disposable {
+public protocol TableViewBond: TableViewBondOptionable {
+	
+	associatedtype DataSource: DataSourceProtocol
+	
+	func apply(event: DataSourceEvent<DataSource>, to tableView: NSTableView)
+	func heightForRow(at index: Int, tableView: NSTableView, dataSource: DataSource) -> CGFloat?
+	func cellForRow(at index: Int, tableView: NSTableView, dataSource: DataSource) -> NSView?
+}
+
+extension TableViewBond {
+	
+	public var insertAnimation: NSTableViewAnimationOptions? {
+		return [.effectFade, .slideUp]
+	}
+	
+	public var deleteAnimation: NSTableViewAnimationOptions? {
+		return [.effectFade, .slideUp]
+	}
+	
+	public func heightForRow(at index: Int, tableView: NSTableView, dataSource: DataSource) -> CGFloat? {
+		return nil
+	}
+	
+	public func apply(event: DataSourceEvent<DataSource>, to tableView: NSTableView) {
```

Comment:
Since we are moving `updating` to the bond object, I guess we can no longer have default method implementation.